spack/package.py

```diff
@@ -108,7 +109,8 @@ class Octopus(AutotoolsPackage, CudaPackage):
         depends_on("libvdwxc~mpi", when="+libvdwxc")
         depends_on("arpack-ng~mpi", when="+arpack")
         depends_on("elpa~mpi", when="+elpa")
-        depends_on("netcdf-fortran ^netcdf-c~~mpi", when="+netcdf")
+        depends_on("netcdf-fortran", when="+netcdf")
```

Comment:
since this line is repeated in both conditions (with and without mpi) its a good idea to push it out of the condition